tests/frameworks/OSXShims/src/LoggingNative.mm

```diff
@@ -15,44 +15,92 @@
 //******************************************************************************
 
 #import <LoggingNative.h>
+#import <Starboard/SmartTypes.h>
 #import <Foundation/NSString.h>
 
 #import <cstdarg>
 
+// NOTE: this function is assumed to only run in Little Endian environments like OSX
+static NSStringEncoding _getWcharEncoding() {
+    switch (sizeof(wchar_t)) {
+        case 2:
+            return NSUTF16LittleEndianStringEncoding;
+        case 4:
+            return NSUTF32LittleEndianStringEncoding;
+        default:
+            assert(false);
+    }
+}
+
+static NSStringEncoding getWcharEncoding() {
```

Comment:
Hm - should we unify on one solution for this on OSX (should we remove my #define from the shim windows.h)?